### DISCUSSION TITLE
Generating own shellcodes for AV bypass

### DISCUSSION THREAD

**Discussion by Pal1Sec** [BishopFox/sliver#708] [Q&A] [answered]
Hi guys! 

First of all thank you for a awesome tool! 

I would like to ask you about generating shellcode to use them into own tools like droppers with AV bypass. Basicaly I need to generate raw shellcode in csharp format, use that in my own encoder, than output from encoder use in my own dropper with AV bypass functionalities. Unfortunatelly I am not able to find working solution. I tried following:

`generate stager -a amd64 --format csharp --lhost <lport> --lport 80 --os windows --protocol http --save shellcode.txt`

But shellcode doesnt work. Could you help me better understand how to prepare custom AV bypass please?

Thank you!

**Comment by moloch--** [comment, marked as answer]
This topic is outside the scope of the Sliver project, there's a lot already written about AV evasion. You'll want to write your own stager or use a packer, `generate stager` simply uses Metasploit to generate the stager code so its likely to still be flagged by AV if you attempt to execute it directly, this code is typically designed for use with exploits or MSF modules.
